### PR TITLE
Ci/refactor rust workflows

### PR DESCRIPTION
**Pull request by apepkuss**

### CODE REVIEW

File: .github/workflows/rust-wasmedge-macro-release.yml [hunk 1]
```diff
@@ -5,62 +5,83 @@ concurrency:
   cancel-in-progress: true
 
 on:
+  # workflow_dispatch:
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-macro-release.yml:8]: Never comments out something in the workflow. If you think this is helpless, then remove it.

File: .github/workflows/rust-wasmedge-macro-release.yml [hunk 2]
```diff
@@ -5,62 +5,83 @@ concurrency:
...
-      - "bindings/rust/wasmedge-macro/**"
-    tags:
-      - "rust-macro/[0-9]+.[0-9]+.[0-9]+*"
-  pull_request:
-    branches:
-      - master
-    paths:
-      - ".github/workflows/rust-wasmedge-macro-release.yml"
-      - "bindings/rust/wasmedge-macro/**"
+      - "ci/**"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-macro-release.yml:13]: You should not only monitor the `ci/**` branches. There is anyone can submit a PR and modify this workflow file.

File: .github/workflows/rust-wasmedge-macro-release.yml [hunk 3]
```diff
@@ -5,62 +5,83 @@ concurrency:
...
-        if: github.ref_type == 'branch'
+      - name: Set git credentials
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+
+      - name: Prepare release
+        run: |
+          cargo install cargo-smart-release
+          # cargo login "$CARGO_REGISTRY_TOKEN"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-macro-release.yml:47]: Do not leave useless line here

File: .github/workflows/rust-wasmedge-sdk-release.yml [hunk 4]
```diff
@@ -5,75 +5,89 @@ concurrency:
   cancel-in-progress: true
 
 on:
+  # workflow_dispatch:
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-sdk-release.yml:8]: Ditto.

File: .github/workflows/rust-wasmedge-sdk-release.yml [hunk 5]
```diff
@@ -5,75 +5,89 @@ concurrency:
...
-    tags:
-      - "rust-sdk/[0-9]+.[0-9]+.[0-9]+*"
-  pull_request:
-    branches:
-      - master
-    paths:
-      - "bindings/rust/**"
-      - "include/api/wasmedge/**"
-      - "lib/api/**"
+      - "ci/**"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-sdk-release.yml:13]: Ditto.

File: .github/workflows/rust-wasmedge-sdk-release.yml [hunk 6]
```diff
@@ -5,75 +5,89 @@ concurrency:
...
 
+      - name: Set git credentials
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+
+      - name: Prepare release
+        run: |
+          cargo install cargo-smart-release
+          # cargo login "$CARGO_REGISTRY_TOKEN"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-sdk-release.yml:47]: Ditto.

File: .github/workflows/rust-wasmedge-sys-release.yml [hunk 7]
```diff
@@ -5,77 +5,89 @@ concurrency:
   cancel-in-progress: true
 
 on:
+  # workflow_dispatch:
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-sys-release.yml:8]: Ditto.

File: .github/workflows/rust-wasmedge-sys-release.yml [hunk 8]
```diff
@@ -5,77 +5,89 @@ concurrency:
...
-    tags:
-      - "rust/[0-9]+.[0-9]+.[0-9]+*"
-  pull_request:
-    branches:
-      - master
-    paths:
-      - "bindings/rust/**"
-      - "include/api/wasmedge/**"
-      - "lib/api/**"
+      - "ci/**"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-sys-release.yml:13]: Ditto.

File: .github/workflows/rust-wasmedge-types-release.yml [hunk 9]
```diff
@@ -5,62 +5,83 @@ concurrency:
   cancel-in-progress: true
 
 on:
+  # workflow_dispatch:
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-types-release.yml:8]: Ditto.

File: .github/workflows/rust-wasmedge-types-release.yml [hunk 10]
```diff
@@ -5,62 +5,83 @@ concurrency:
...
-      - "bindings/rust/wasmedge-types/**"
-    tags:
-      - "rust-types/[0-9]+.[0-9]+.[0-9]+*"
-  pull_request:
-    branches:
-      - master
-    paths:
-      - ".github/workflows/rust-wasmedge-types-release.yml"
-      - "bindings/rust/wasmedge-types/**"
+      - "ci/**"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-types-release.yml:13]: Ditto.

File: .github/workflows/rust-wasmedge-types-release.yml [hunk 11]
```diff
@@ -5,62 +5,83 @@ concurrency:
...
-        if: github.ref_type == 'branch'
+      - name: Set git credentials
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+
+      - name: Prepare release
+        run: |
+          cargo install cargo-smart-release
+          # cargo login "$CARGO_REGISTRY_TOKEN"
```
Comment (hydai) [inline .github/workflows/rust-wasmedge-types-release.yml:47]: Ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by apepkuss** [comment]
@hydai Please ignore this PR as I made it from a wrong branch. I split the changes in this PR into two separate PRs, #1964 and #1965. The comments above should be related to PR #1964, so I'll check and update the relevant files in PR #1964 and close this PR. Sorry for the mistake and thanks for the review!

**Pull request closed without merging**